src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/IAggregateValidator.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    /// <summary>
+    /// Runs a set of <see cref="IValidator"/>s.
+    /// </summary>
+    public interface IAggregateValidator
+    {
+        /// <summary>
+        /// Human readable name of this <see cref="IAggregateValidator"/>.
+        /// </summary>
+        string Name { get; }
+
+        /// <summary>
+        /// Runs each <see cref="IValidator"/> and returns an <see cref="AggregateValidationResult"/> containing all results.
+        /// </summary>
+        Task<AggregateValidationResult> ValidateAsync(ValidationContext data);
+    }
+}
```

Comment:
Actually wondering whether IAggregateValidator and AggregateValidationResult are necessary. It looks and behaves very similar to IValidator, except for returning an enumerable of results. Do we really need to differentiate the aggregator?

Example of this would be the AggregateAuditingService in Gallery, which implements the same interface as the individual auditing services.